src/Form/EntityEmbedDialog.php

```diff
@@ -737,6 +736,11 @@ public function submitEmbedStep(array &$form, FormStateInterface $form_state) {
         $values['attributes']['data-entity-embed-settings'] = Json::encode($values['attributes']['data-entity-embed-settings']);
       }
 
+      // Filter out empty attributes.
+      $values['attributes'] = array_filter($values['attributes'], function($value) {
```

Comment:
It would filter out an alt text or caption value of "0", so I figured strlen was safer in this case.